py-polars/tests/unit/namespaces/test_datetime.py

```diff
@@ -897,3 +897,62 @@ def test_median(values: list[date | None], expected_median: date | None) -> None
 def test_mean(values: list[date | None], expected_mean: date | None) -> None:
     result = pl.Series(values).cast(pl.Date).dt.mean()
     assert result == expected_mean
+
+
+@pytest.mark.parametrize(
+    ("left_time_unit", "right_time_unit", "expected_time_unit"),
+    [
+        ("ms", "ms", "ms"),
+        ("ms", "us", "us"),
+        ("us", "ms", "us"),
+        ("us", "us", "us"),
+        ("ms", "ns", "ns"),
+        ("us", "ns", "ns"),
+        ("ns", "ms", "ns"),
+        ("ns", "us", "ns"),
+        ("ns", "ns", "ns"),
+    ],
+)
+def test_subtract_datetimes_keep_subsecond_precision(
```

Comment:
@MarcoGorelli in a very similar fashion to the other test, but with subtraction between datetimes of different precisions instead of summing a duration

Reply:
nice!